conda-store-server/conda_store_server/build.py

```diff
@@ -205,9 +203,7 @@ def solve_conda_environment(conda_store, solve):
 
             channel_id = api.get_conda_channel(conda_store.db, channel)
             if channel_id is None:
-                raise ValueError(
-                    f"channel url={channel} not recognized in conda-store channel database"
-                )
+                channel_id = api.add_conda_channel(conda_store.db, channel)
```

Comment:
Need a conditional check `if conda_store.default_allow_channels: ...` here because ValueError should be raised if not enabled.

Reply:
Ok, I think I was under the assumption that since the channel list as part of the specification has already been validated, that this is no longer necessary.

Reply:
Realized that this make much sense. Done. ✅